src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -464,14 +465,21 @@ object Parser {
         "pure" <~ eos ^^^ PureClause
       }
 
-      functSpecClause.* ~ "pure".? ^^ {
+      val specClauses = if(withTrusted) {
+        functSpecClause | "trusted" <~ eos ^^^ TrustedClause
+      } else {
+        functSpecClause
+      }
+
+      specClauses.* ~ "pure".? ^^ {
```

Comment:
we can revert the changes in this line after incorporating my previous suggestion.